### DISCUSSION TITLE
@gym conditional equipment

### DISCUSSION THREAD

**Discussion by yrinkel** [astashov/liftosaur#153] [Ideas]
When rotating between different gyms, some are simply missing certain types of equipment. 
Same for home gym, you probably don’t have a full set of machine, so you’d like to use a certain machine at the commercial gym and fallback to your dumbbells or whatever at your home gym. 

Currently, that requires manual editing of the script every time (multiple times a week for me when switching back and forth). 

It would be great to be able to conditionally define and automatically select an equipment based on the “current gym” setting. Something like that:

```
@home: Bench Press, Dumbell / 3x12 /
@fitness24: Bench Press, Barbell / 3x12 /
```
Or 
```
@GoldsGym: Squat, Barbell / 3x8 /
@HotelGym: Squat, Smith Machine / 3x8 /
```

**Comment by yrinkel** [comment]
> Astashov correctly points out that @gym syntax can’t be shared as Liftosaur knows nothing about gyms. 

Yes what you say makes sense.

Then maybe a way to deactivate default equipment in some gyms, to let Liftosaur know that this equipment is not available in the currently selected gym? That could be coupled with an alternative equipment setting, where you can define an alternative in the equipment’s setting in case it’s not available.

For example, if Barbell is not available use Smith Machine, and if that not available use Dumbbell

**Comment by wozers** [comment]
I highly agree.

**Comment by buhund** [comment]
Another use case is the fact that no two gyms keep the same cable machine and no two cable machines keep the same weight intervals, and even if you somehow get more or less the same number, no two cable machines will have the same resistance on the same weight (which seems to be some law of nature).

Being able to select gym, and from that retrieve the previous weights used for each exercise for progression, would make life for us multi-gym users a whole lot better. Especially relevant for cable machines.

**Comment by hcars** [comment]
I think a great extension to this would be allow users to crowdsource gym equipment as an option.
